libraries/lib-import-export/EditRiffTags.cpp

```
@@ -0,0 +1,115 @@
+   // loop is one-shot:
+   auto type = reinterpret_cast<uint32_t*>(dataUPtr.get());
+   if (acidTags.isOneShot)
+      *type |= 0x00000001;
+
+   // Set the meter denominator 2 bytes to 4:
+   auto numerator = reinterpret_cast<uint16_t*>(dataUPtr.get() + 16);
+   *numerator |= 0x0004;
+   auto denominator = reinterpret_cast<uint16_t*>(dataUPtr.get() + 18);
+   *denominator |= 0x0004;
+
+   auto tempo = reinterpret_cast<float*>(dataUPtr.get() + 20);
+   *tempo = acidTags.bpm;
+
+   const auto result = sf_set_chunk(file, &chunk);
+   assert(result == SF_ERR_NO_ERROR);
+}
+
+void AddDistributorInfo(const std::string& distributor, SNDFILE* file)
+{
+   const int32_t distributorSize = distributor.size();
+   SF_CHUNK_INFO chunk;
+   snprintf(chunk.id, sizeof(chunk.id), "LIST");
+   chunk.id_size = 4;
+   constexpr std::array<char, 4> listTypeID = { 'I', 'N', 'F', 'O' };
+   constexpr std::array<char, 4> distributorTypeID = { 'I', 'D', 'S', 'T' };
+   chunk.datalen = sizeof(listTypeID) + sizeof(distributorTypeID) +
+                   sizeof(distributorSize) + distributorSize;
+   // A trick taken from libsndfile's source code, probably to ensure that
+   // the rest of the data stays word-aligned:
+   while (chunk.datalen & 3)
+      ++chunk.datalen;
+   const auto dataUPtr = std::make_unique<char[]>(chunk.datalen);
+   chunk.data = dataUPtr.get();
+   memset(chunk.data, 0, chunk.datalen);
+   char* chars = reinterpret_cast<char*>(chunk.data);
+   auto pos = 0;
+
+   memcpy(chars, listTypeID.data(), sizeof(listTypeID));
+
+   pos += sizeof(listTypeID);
+   memcpy(chars + pos, distributorTypeID.data(), sizeof(distributorTypeID));
+
+   pos += sizeof(distributorTypeID);
+   memcpy(chars + pos, &distributorSize, sizeof(distributorSize));
+
+   pos += sizeof(distributorSize);
+   memcpy(chars + pos, distributor.c_str(), distributor.size());
+
+   const auto result = sf_set_chunk(file, &chunk);
+   assert(result == SF_ERR_NO_ERROR);
```

Why not propagate the error out of this function and let the caller decide whether to eat it?

---

I don't expect this to be an issue beyond development of this utility, and am lazy handling it on the client side.